File: Modelos Java/ArrayJava/src/main/java/com/mycompany/arrayjava/Modelo/Modelo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.arrayjava.Modelo;

/**
 *
 * @author 57320
 */

import java.util.ArrayList;
import java.util.Random;

public class Modelo {
    public ArrayList<Politico> politicos = new ArrayList<>();

    public void generarPoliticos(int n) {
        Random rand = new Random();
        politicos.clear();
        for (int i = 0; i < n; ++i) {
            Politico p = new Politico();
            p.id = i + 1;
            p.dineroRobado = rand.nextInt(1000000) + 100;
            p.fechaNacimiento = generarFechaAleatoria(rand);
            politicos.add(p);
        }
    }

    public void ordenarPoliticos(String metodo) {
        if (metodo.equalsIgnoreCase("bubble")) {
            bubbleSort();
        } else if (metodo.equalsIgnoreCase("selection")) {
            selectionSort();
        } else {
            System.out.println("Metodo de ordenamiento invalido. Se usara Bubble Sort por defecto.");
            bubbleSort();
        }
    }

    private void bubbleSort() {
        boolean swapped;
        for (int i = 0; i < politicos.size() - 1; i++) {
            swapped = false;
            for (int j = 0; j < politicos.size() - i - 1; j++) {
                if (politicos.get(j).dineroRobado > politicos.get(j + 1).dineroRobado) {
                    Politico temp = politicos.get(j);
                    politicos.set(j, politicos.get(j + 1));
                    politicos.set(j + 1, temp);
                    swapped = true;
                }
            }
            if (!swapped) break;
        }
    }

    private void selectionSort() {
        int n = politicos.size();
        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (politicos.get(j).dineroRobado < politicos.get(minIndex).dineroRobado) {
                    minIndex = j;
                }
            }
            Politico temp = politicos.get(i);
            politicos.set(i, politicos.get(minIndex));
            politicos.set(minIndex, temp);
        }
    }

    private String generarFechaAleatoria(Random rand) {
        int dia = rand.nextInt(28) + 1;
        int mes = rand.nextInt(12) + 1;
        int anio = rand.nextInt(50) + 1950;
        return dia + "/" + mes + "/" + anio;
    }
}

File: Modelos Python/ArrayPython/modelo/modelo.py
import random

class Politico:
    def __init__(self, id, dinero_robado, fecha_nacimiento):
        self.id = id
        self.dinero_robado = dinero_robado
        self.fecha_nacimiento = fecha_nacimiento

class Modelo:
    def __init__(self):
        self.politicos = []

    def generar_politicos(self, n):
        random.seed()
        self.politicos = []
        for i in range(n):
            id = i + 1
            dinero_robado = random.randint(100, 1000000)
            fecha_nacimiento = self.generar_fecha_aleatoria()
            self.politicos.append(Politico(id, dinero_robado, fecha_nacimiento))

    def ordenar_politicos(self, metodo):
        if metodo == "bubble":
            self.bubble_sort()
        elif metodo == "selection":
            self.selection_sort()
        else:
            print("Metodo de ordenamiento invalido. Se usara Bubble Sort por defecto.")
            self.bubble_sort()

    def bubble_sort(self):
        n = len(self.politicos)
        for i in range(n - 1):
            for j in range(n - i - 1):
                if self.politicos[j].dinero_robado > self.politicos[j + 1].dinero_robado:
                    self.politicos[j], self.politicos[j + 1] = self.politicos[j + 1], self.politicos[j]

    def selection_sort(self):
        n = len(self.politicos)
        for i in range(n - 1):
            min_index = i
            for j in range(i + 1, n):
                if self.politicos[j].dinero_robado < self.politicos[min_index].dinero_robado:
                    min_index = j
            self.politicos[i], self.politicos[min_index] = self.politicos[min_index], self.politicos[i]

    def generar_fecha_aleatoria(self):
        dia = random.randint(1, 28)
        mes = random.randint(1, 12)
        anio = random.randint(1950, 1999)
        return f"{dia}/{mes}/{anio}"
